cpu/stm32/dist/irqs/gen_irqs.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2020 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+"""Generate a header with the number of IRQs for a given STM32 family."""
+
+import os
+import argparse
+import re
+
+
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+RIOTBASE = os.getenv(
+    "RIOTBASE", os.path.abspath(os.path.join(CURRENT_DIR, "../../../..")))
+STM32_INCLUDE_DIR = os.path.join(RIOTBASE, "cpu/stm32/include")
+STM32_CMSIS_INCLUDE_DIR = os.path.join(
+    RIOTBASE, STM32_INCLUDE_DIR, "vendor/cmsis/{}/Include/")
+STM32_CMSIS_HEADER = os.path.join(
+    RIOTBASE, STM32_CMSIS_INCLUDE_DIR, "{}.h")
+STM32_IRQS_DIR = os.path.join(
+    RIOTBASE, STM32_INCLUDE_DIR, "irqs/{}/irqs.h")
+
+IRQS_FORMAT = """
+/*
+ * PLEASE DON'T EDIT
+ *
+ * This file was automatically generated by
+ * ./cpu/stm32/dist/irqs/gen_irqs.py
+ */
+
+#ifndef IRQS_{cpu_fam}_H
+#define IRQS_{cpu_fam}_H
+
+#ifdef __cplusplus
+extern "C" {{
+#endif
+
+{irq_lines}
+
+#ifdef __cplusplus
+}}
+#endif
+
+#endif /* IRQS_{cpu_fam}_IRQ_H */
+"""
```

Comment:
Better put this in a different file no?

Reply:
The string is used as a template for `format`. No need to add extra complexity to store it in a separate file. The initial version was based on Jinja2 for which it made sense but now I don't see any benefit.